source/scripts/Background/controllers/MessageHandler/requests.ts

```diff
@@ -1,276 +1,307 @@
+};
 
 export const handleRequest = async (
-    port: Runtime.Port,
-    masterController: IMasterController,
-    message: Message,
-    origin: string,
-    setPendingWindow: (isPending: boolean) => void
+  port: Runtime.Port,
+  masterController: IMasterController,
+  message: Message,
+  origin: string,
+  setPendingWindow: (isPending: boolean) => void
 ) => {
-    const { vault } = store.getState();
-    const allWallets = [...vault.wallets.local, ...vault.wallets.ledger];
-    const activeWallet: KeyringWalletState | null = 
-        vault?.activeWallet ? allWallets.find(
-            (wallet: any) => wallet.id === vault.activeWallet.id
-        ) : null;
-    const { method, args, asset } = message.data;
-
-    console.log('CAL_REQUEST.method:', method, args);
-
-    const allowed = masterController.dapp.isDAppConnected(origin);
-    const walletIsLocked = !masterController.wallet.isUnlocked();
-    const provider = asset === 'DAG' ? masterController.stargazerProvider : masterController.ethereumProvider;
-    const network = asset === 'DAG' ? KeyringNetwork.Constellation : KeyringNetwork.Ethereum;
-    const windowUrl = activeWallet.type ===  KeyringWalletType.LedgerAccountWallet ? LEDGER_URL : EXTERNAL_URL;
-    const windowType = activeWallet.type === KeyringWalletType.LedgerAccountWallet ? WINDOW_TYPES.normal : WINDOW_TYPES.popup;
-    const windowSize = activeWallet.type === KeyringWalletType.LedgerAccountWallet ? { width: 1000, height: 1000 } : { width: 372, height: 600 }
-    const windowId = uuid();
-
-
-    let result: any = undefined;
-    switch (+method) {
-        case SUPPORTED_WALLET_METHODS.isConnected:
-            result = { connected: !!allowed && !walletIsLocked };
-            break;
-        case SUPPORTED_WALLET_METHODS.getAddress:
-            result = provider.getAddress();
-            break;
-        case SUPPORTED_WALLET_METHODS.getAccounts:
-            result = provider.getAccounts();
-            break;
-        case SUPPORTED_WALLET_METHODS.getChainId:
-            result = provider.getChainId();
-            break;
-        case SUPPORTED_WALLET_METHODS.getBlockNumber:
-            result = provider.getBlockNumber();
-            break;
-        case SUPPORTED_WALLET_METHODS.estimateGas:
-            result = await provider.getGasEstimate();
-            break;
-        case SUPPORTED_WALLET_METHODS.getNetwork:
-            result = provider.getNetwork();
-            break;
-        case SUPPORTED_WALLET_METHODS.getBalance:
-            result = provider.getBalance();
-            break;
-        case SUPPORTED_WALLET_METHODS.signMessage:{
-            if(!activeWallet){
-                return Promise.reject(new CustomEvent(message.id, {
-                    detail: 'There is no active wallet'
-                }));
-            }
+  const { vault } = store.getState();
+  const allWallets = [...vault.wallets.local, ...vault.wallets.ledger];
```

Comment:
Can we make this a lib function somewhere? Cesar is about to add a 3rd type so it'd be easier to update if it's in a central place